es_ES/faq.md

```diff
@@ -0,0 +1,625 @@
+
+* Programación declarativa: los métodos de compilación de widgets suelen ser una única expresión con múltiples niveles de constructores anidados, escritos utilizando un subconjunto declarativo de Dart. Tales expresiones anidadas podrían ser mecánicamente transformado hacia o desde cualquier lenguaje de markup expresivo adecuado. Por ejemplo, el widget [UserAccountsDrawerHeader](https://docs.flutter.io/flutter/material/UserAccountsDrawerHeader-class.html) tiene un método de compilación larga (más de 20 líneas), que consiste en una sola expresión anidada. Esto también se puede combinar con el estilo imperativo para construir UI eso sería más difícil de describir en un enfoque puramente declarativo.
+
+* Programación genérica: los tipos pueden usarse para ayudar a los desarrolladores a captar errores tempranos en la programación. El framework de Flutter utiliza programación genérica para ayudar en esto. Por ejemplo, la clase [State](https://docs.flutter.io/flutter/widgets/State-class.html) es parametrizada en términos del tipo de su widget asociado, de modo que el analizador Dart puede detectar desajustes entre estados y widgets. Del mismo modo, la clase [GlobalKey](https://docs.flutter.io/flutter/widgets/GlobalKey-class.html) toma un parámetro de tipo para que pueda acceder al estado de un widget remoto en un tipo de manera segura (utilizando la comprobación de tiempo de ejecución), la interfaz [Route](https://docs.flutter.io/flutter/widgets/Route-class.html) es parametrizada con el tipo que se espera que use cuando [popped](https://docs.flutter.io/flutter/widgets/Navigator/pop.html), y colecciones tales como [List](https://docs.flutter.io/flutter/dart-core/List-class.html), [Map](https://docs.flutter.io/flutter/dart-core/Map-class.html), y [Set](https://docs.flutter.io/flutter/dart-core/Set-class.html) todas son parametrizadas de modo que los elementos no coincidentes se puedan detectar temprano durante análisis o en tiempo de ejecución durante la depuración.
+
+* Programación concurrente: Flutter hace un uso intensivo de [Future](https://docs.flutter.io/flutter/dart-async/Future-class.html) y otras API asíncronas. Por ejemplo, el sistema de animación informa cuando la animación se completa al completar un futuro. El sistema de carga de imagen de manera similar utiliza futuros para informar cuando se completa una carga.
+
+* Programación de restricción: el sistema de disposición en Flutter usa una forma débil de programación de restricciones para determinar la geometría de una escena. Restricciones (para cajas cartesianas, un ancho mínimo y máximo y un mínimo y máximo altura) se pasan de padres a hijos, y el hijo selecciona una geometría resultante (por ejemplo, para cajas cartesianas, un tamaño, específicamente un ancho y una altura) que cumple con esas restricciones. Al usar esta técnica, Flutter generalmente puede diseñar una escena completa con un solo pase.
```

Comment:
(e.g., for cartesian boxes, a size, specifically a width and a height) 
(ej., para cajas cartesianas, una dimensión, escifícamente un ancho y un alto)